drift_dev/lib/src/writer/manager/manager_templates.dart

```diff
@@ -0,0 +1,387 @@
+part of "database_manager_writer.dart";
+
+/// A class which contains utility functions to generate manager class names
+///
+/// This is used by the [DatabaseManagerWriter] to generate code for the manager classes
+class _ManagerCodeTemplates {
+  _ManagerCodeTemplates(this.__scope);
+
+  /// A Scope class which contains the current scope of the generation
+  ///
+  /// Used to generating names which require import prefixes
+  final Scope __scope;
+
+  /// Returns the name of the manager class for a table
+  ///
+  /// This classes acts as container for all the table managers
+  ///
+  /// E.g. `AppDatabaseManager`
+  String databaseManagerName(String dbClassName) {
+    return '${dbClassName}Manager';
+  }
+
+  /// How the database will represented in the generated code
+  ///
+  /// When doing modular generation the table doesnt have direct access to the database class
+  /// so it will use `GeneratedDatabase` as the generic type in such cases
+  ///
+  /// E.g. `i0.GeneratedDatabase` or `AppDatabase`
+  String databaseType(TextEmitter leaf, String dbClassName) {
+    return switch (__scope.generationOptions.isModular) {
+      true => leaf.drift("GeneratedDatabase"),
+      false => dbClassName,
+    };
+  }
+
+  /// The code for the database manager getter which will be added to the main database class
+  ///
+  /// E.g. `AppDatabase get managers => AppDatabaseManager(this);`
+  String databaseManagerGetter(String dbClassName) {
```

Comment:
I haven't checked, but this sounds like something we'll only ever use at a single place, right (when generating the main database class)? In that case I think it's easier to define it in the class using it rather than moving it out into a second file (I agree about this refactoring for stuff called in different places though).

This applies to some of the other definitions below writing getters as well - IMO it's easier to have that all in one class if it's only used by one class. The methods returning names/references should be here, but if things aren't used multiple times I think separating them just makes code harder to follow.

Reply:
I'll move the smaller code templates that are only used once.
The larger code templates pollute the logic of the actual generation and I think should remain in the templates class, what are your thoughts?